File: be-ojt2026/Services/Inventory/Inventory.Domain/Entities/Warehouse.cs
using System;
using System.Collections.Generic;

namespace Inventory.Domain.Entities;

public partial class Warehouse
{
    public int Id { get; set; }

    public string WarehouseCode { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string? Address { get; set; }

    public string? Phone { get; set; }

    public string? Email { get; set; }

    public int? ManagerId { get; set; }

    public string? WarehouseType { get; set; }

    public decimal? AreaSqm { get; set; }

    public bool? IsActive { get; set; }

    public string? Notes { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<Location> Locations { get; set; } = new List<Location>();

    public virtual ICollection<ProductWarehouse> ProductWarehouses { get; set; } = new List<ProductWarehouse>();
}

File: fe-ojt2026/src/types/warehouse/warehouse.ts
// Types for warehouse structure APIs.

export interface Location {
  Id: number;
  Code: string;
  Name: string;
  Type: string;
  Address?: string;
  WarehouseId: number;
  IsActive: boolean;
  ParentId?: number;
}

export interface Carrier {
  id: number;
  name: string;
  phone: string;
}

export interface Warehouse {
  Id: number;
  WarehouseCode: string;
  Name: string;
  Address: string | null;
  Phone: string | null;
  Email: string | null;
  ManagerId: number | null;
  WarehouseType: string | null;
  AreaSqm: number | null;
  IsActive: boolean;
  Notes: string | null;
  CreatedAt: string | null;
  UpdatedAt: string | null;
}
